### ISSUE TITLE
Node Monitor Not Getting Domain Suffix, Thereby Not Sending Email Notifications

### ISSUE THREAD

**Issue by consentfactory** [netdisco/netdisco#783]
## Expected Behavior
When Netdisco node monitor detects a MAC address, an email via sendmail should be sent. 

## Current Behavior
Email is not sent. Running `bin/netdisco-do -D nodemonitor` to detect a MAC address, the following error in the terminal appears:

```
Use of uninitialized value $domain in pattern match (m//) at /home/netdisco/perl5/lib/perl5/App/Netdisco/Util/NodeMonitor.pm line 20.
Use of uninitialized value $domain in concatenation (.) or string at /home/netdisco/perl5/lib/perl5/App/Netdisco/Util/NodeMonitor.pm line 25.
```

Looking at the info in /var/spool/mail/<user>, email isn't being sent due to missing 'From' information. 

```
The original message was received at Tue, 13 Apr 2021 14:00:53 -0600
from localhost [127.0.0.1]

   ----- The following addresses had permanent fatal errors -----
<consentfactory@domain.com>
    (reason: 550-5.7.1 [xxx.xxx.xxx.xxx     14] Messages missing a valid address in From:)
```

## Possible Solution
My temporary work around is to manually configure the domain suffix in `/home/netdisco/perl5/lib/perl5/App/Netdisco/Util/NodeMonitor.pm (line 20)`:

```
 20   #my $domain =~ (hostfqdn || 'fqdn-undefined');
 21   my $domain = 'domain.com';
``` 

After which, the emails work as expected.

## Steps to Reproduce (for bugs)
1. Configure node to monitor in 'Node Monitor'
2. Force run node monitor through terminal

## Context
Send email notifications about nodes appearing. 

I have verified `/etc/hosts` and `/etc/hostname` is configured correctly. 

I have also ran multiple manual sendmail events without issue to our relay. 

## Your Environment
* OS: Red Hat Enterprise Linux Server release 7.9 (Maipo)
* Netdisco version used: 2.47.5
* SNMP::Info version used: 3.71
* Perl: 5.16.3
* Sendmail: 8.14.7

## Config info (deployment.yml)
Most settings are default. I've tried adding this:

```
NETDISCO_DOMAIN: 'domain.com'

# will be stripped from fqdn when displayed in the web UI
# also, do not forget the leading dot.
# ```````````````````````````````````````````````````````
domain_suffix: '.domain.com'
```

Let me know what else you may need and I can grab the info.

Thanks.

**Comment by inphobia** [comment]
hm, it seems like

https://github.com/netdisco/netdisco/blob/951139fe7c6511c344bd20b2543463a6828ff295/lib/App/Netdisco/Util/NodeMonitor.pm#L20

should be
`my $domain = (hostfqdn || 'fqdn-undefined'); `
instead.

could you try that change & report back?

it seems this change came from transforming `domain_suffix` to a list, but i don't see a reason why that should have been done for the email from domain.

(#452)

**Comment by ollyg** [comment]
thanks, fixed in ce29cf2d
